src/features/channels/components/message-reply-provider/MessageReplyProvider.tsx

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+import { createContext, ReactElement, useMemo, useState } from 'react';
+
+type ReplyType = {
+  message: string;
+  name: string;
+};
+
+export type MessageReplyContextType = {
```

Comment:
I like the React context implementation. There is one question though. Isn't the slice we already have enough to accomplish the same functionality? I think that if we mix context and redux to accomplish the same things we can end up with a mix of styles in the app. My suggestion is to use the redux state to share the reply state to other components and keep using react context to do smaller things globally like switching theme, etc.